src/nunit.analyzers.tests/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzerTests.cs

```diff
@@ -47,45 +49,72 @@ class MyTests
         }
 
         [Test]
-        public void AnalyzeWhenStringConstant()
+        public void FixWhenStringLiteral()
         {
-            var expectedDiagnostic = ExpectedDiagnostic.Create(
-                AnalyzerIdentifiers.TestCaseSourceStringUsage,
-                String.Format(TestCaseSourceUsageConstants.ConsiderNameOfInsteadOfStringConstantMessage, "Tests"));
-
             var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
     public class AnalyzeWhenStringConstant
     {
-        [↓TestCaseSource(""Tests"")]
+        private static readonly TestCaseData[] TestCases = new TestCaseData[0];
+
+        [TestCaseSource(↓""TestCases"")]
+        public void Test()
+        {
+        }
+    }");
+
+            var fixedCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    public class AnalyzeWhenStringConstant
+    {
+        private static readonly TestCaseData[] TestCases = new TestCaseData[0];
+
+        [TestCaseSource(nameof(TestCases))]
         public void Test()
         {
         }
     }");
-            AnalyzerAssert.Diagnostics(analyzer, expectedDiagnostic, testCode);
+            var message = "Consider using nameof(TestCases) instead of \"TestCases\"";
+            AnalyzerAssert.CodeFix(Analyzer, Fix, ExpectedDiagnostic.WithMessage(message), testCode, fixedCode, allowCompilationErrors: AllowCompilationErrors.Yes);
```

Comment:
Not sure why the fix introduces compilation errors here, using `allowCompilationErrors: AllowCompilationErrors.Yes` to work around it.

Reply:
Yes `AnalyzerAssert.CodeFix` the problem is that the fixed code reports a compiler error. I'll try rewriting it using `SyntaxFactory.ParseExpression($"nameof({valueText})")`